Java/core/src/main/java/com/amazon/randomcutforest/store/PointStoreFloat.java

```diff
@@ -115,27 +97,23 @@ public boolean pointEquals(int index, float[] point) {
      * @return a copy of the point stored at the given index.
      * @throws IllegalArgumentException if the index value is not valid.
      * @throws IllegalArgumentException if the current reference count for this
-     *                                  index is non-positive.
+     *                                  index is nonpositive.
      */
     @Override
     public float[] get(int index) {
         checkValidIndex(index);
-        return Arrays.copyOfRange(store, index * dimensions, (index + 1) * dimensions);
+        int address = (directLocationMap) ? index * dimensions : locationList[index];
+        return Arrays.copyOfRange(store, address, address + dimensions);
     }
 
-    /**
-     * print the point at location index
-     * 
-     * @param index index of the point in the store
-     * @return ascii output
-     */
     @Override
     public String toString(int index) {
         return Arrays.toString(get(index));
     }
 
-    public float[] getStore() {
-        return store;
+    @Override
+    void copyTo(int current, int running) {
```

Comment:
nit: maybe change the formal method arguments to make the direction of the copy clear:
```
void copyTo(int dest, int source)
```

Reply:
Done. Changed it to copy as a block.